File: backend/app/schemas/sensor.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class SensorLogCreate(BaseModel):
    ph: float
    tds: float
    water_level: float
    dht_temp: float
    dht_humidity: float
    water_temp: float
    ph_reducer: str = "OFF"
    add_water: Optional[str] = "OFF"
    nutrients_adder: str = "OFF"
    humidifier: str = "OFF"
    ex_fan: str = "OFF"

class SensorLogOut(SensorLogCreate):
    id: int
    timestamp: datetime
    class Config:
        from_attributes = True

File: frontend/src/types/sensor.ts
export interface SensorReading {
  id: number
  ph: number
  tds: number
  water_level: number
  dht_temp: number
  dht_humidity: number
  water_temp: number
  ph_reducer: 'ON' | 'OFF'
  add_water: 'ON' | 'OFF'
  nutrients_adder: 'ON' | 'OFF'
  humidifier: 'ON' | 'OFF'
  ex_fan: 'ON' | 'OFF'
  timestamp: string
}
